addons/project/models/mail_thread.py

```diff
@@ -0,0 +1,17 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+from odoo.addons.project.controllers.project_sharing_chatter import ProjectSharingChatter
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    @api.model
+    def _get_thread_with_access(self, thread_id, mode="read", **kwargs):
```

Comment:
``_get_thread_with_access`` is a model method, called on the right model. This override should therefore be done only on project and/or task model, this is irrelevant as a global mixin override.